src/hashtree.erl

```diff
@@ -413,14 +457,21 @@ width(#state{width=W}) ->
 mem_levels(#state{mem_levels=M}) ->
     M.
 
+-spec path(hashtree()) -> string().
+path(#state{path=P}) ->
+    P.
+
 %% Note: meta is currently a one per file thing, even if there are multiple
 %%       trees per file. This is intentional. If we want per tree metadata
 %%       this will need to be added as a separate thing.
--spec write_meta(binary(), binary(), hashtree()) -> hashtree().
+-spec write_meta(binary(), binary()|term(), hashtree()) -> hashtree().
 write_meta(Key, Value, State) when is_binary(Key) and is_binary(Value) ->
     HKey = encode_meta(Key),
     ok = eleveldb:put(State#state.ref, HKey, Value, []),
-    State.
+    State;
+write_meta(Key, Value0, State) when is_binary(Key) ->
```

Comment:
I'd say fraught with peril.  I think we've been trying to fix the library in ways it wasn't quite intended.  hashtree is used as a library by kv_index_hashtree/yz_index-hashtree, and it's those two containers that know if it's shutdown correctly.

We should probably fix by providing some functions for manipulating the open/closed counts and setting the type of rebuild.

```mark_open_and_check(TreeId, State)``` - compare opened/closed, set rebuild type, increment opened, 
```mark_clean_close(TreeId, State)``` - set closed to opened.